DigitalLearningSolutions.Web/Views/Frameworks/Shared/_FrameworkPageSearch.cshtml

```diff
@@ -1,13 +1,13 @@
-﻿@using DigitalLearningSolutions.Web.ViewModels.Common.SearchablePage
-@model BaseSearchablePageViewModel
-
-<div class="search-box-container" id="search">
-  <label class="nhsuk-u-visually-hidden" for="search-field">Search frameworks</label>
-  <input class="search-box" id="search-field" name="searchString" type="search" placeholder="Search" autocomplete="off" value="@Model.SearchString">
-  <button class="nhsuk-search__submit" type="submit" asp-action="@ViewContext.RouteData.Values["action"].ToString()" asp-route-page="1">
-    <svg class="nhsuk-icon nhsuk-icon__search" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" aria-hidden="true" focusable="false">
-      <path d="M19.71 18.29l-4.11-4.1a7 7 0 1 0-1.41 1.41l4.1 4.11a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42zM5 10a5 5 0 1 1 5 5 5 5 0 0 1-5-5z"></path>
-    </svg>
-    <span class="nhsuk-u-visually-hidden">Search</span>
-  </button>
-</div>
+﻿@using DigitalLearningSolutions.Web.ViewModels.Common.SearchablePage
+@model IBaseSearchablePageViewModel

```

Comment:
Given that BaseSearchablePageViewModel<T> only takes T : BaseSearchableItem, what if you use `@model BaseSearchablePageViewModel<BaseSearchableItem>`? Covariance should get you there. As in https://stackoverflow.com/a/7299071

Reply:
It might be possible with a more close look at it, but when I just tried to do this simply, I get build errors converting e.g. a `CentreAdministratorsViewModel : BaseSearchablePageViewModel<AdminUser>` to a `BaseSearchablePageViewModel<BaseSearchableItem>` despite `AdminUser` implementing it via `User`

Reply:
I also can't really do this with the BasePaginatedViewModel as it will break compatibility with the Obsolete version

Reply:
Ah sorry, you can only make an interface covariant. So I don't think it makes it any better, you'd still have to use an interface.
But in case you're interested academically: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/out-generic-modifier